app/src/lib/git/status.ts

```diff
@@ -35,7 +43,18 @@ export interface IStatusResult {
   readonly workingDirectory: WorkingDirectoryStatus
 }
 
-function convertToAppStatus(status: FileEntry): AppFileStatus {
+type StatusHeadersData = {
+  currentBranch: string | undefined
```

Comment:
All of these that have the type signature `x | undefined` can be made optional `x?: string`